### PR TITLE
Create c-tag-find/c-today/c-tomorrow feature

### LINKED ISSUE #138: As a store manager, I want to know who are working today
... so that I can easily distribute the work among the available employees.

### LINKED ISSUE #139: As a store manager, I want to know who are working on the next day
... so that I can make plan for next day beforehand.

### LINKED ISSUE #140: As an advanced user, I can conveniently get a filtered group of people whose tag(s) contain the keyword(s) I enter
... so that I can easily find out the employee(s) who belongs to the same category/categories.

### PR DESCRIPTION
**Pull request by yanbingtao**
1. `c-today`:
- Able to list down all persons whose tag(s) match today's day (i.e. Monday, Tuesday, etc).
2. `c-tomorrow`:
- Able to list down all persons whose tag(s) match the next day's day (i.e. Monday, Tuesday, etc).
3. `c-tag-find KEYWORD [KEYWORDS] ...`
- Able to list down all persons whose tag(s) match the keyword(s) entered by the user.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.TagContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TagFindCommand.java]: Incorrect javadoc? I'm reviewing based on the assumption that this command will find an employee by the tag.

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class TagFindCommand extends Command {
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TagFindCommand.java]: I think `FindTagCommand`or `FindByTagCommand`  would be more apt/ sounds better.

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    public static final String COMMAND_WORD = "c-tag-find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TagFindCommand.java]: Same problem as the javadoc

File: src/main/java/seedu/address/logic/commands/TodayCommand.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+
+import seedu.address.model.Model;
+
+public class TodayCommand extends Command {
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TodayCommand.java]: So this is a different feature from the one listed in this PR?

File: src/main/java/seedu/address/logic/commands/TodayCommand.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            break;
+        case FRIDAY:
+            model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_FRIDAY_PERSONS);
+            break;
+        case SATURDAY:
+            model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_SATURDAY_PERSONS);
+            break;
+        default:
+            model.updateFilteredPersonList(person -> false);
+
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TodayCommand.java]: Remove extra line.

File: src/main/java/seedu/address/logic/commands/TodayCommand.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        if (model.getFilteredPersonList().size() == 0) {
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TodayCommand.java]: I think it is better to add a blank line above this if-else block

File: src/main/java/seedu/address/logic/commands/TomorrowCommand.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        default:
+            model.updateFilteredPersonList(person -> false);
+
+        }
+        if (model.getFilteredPersonList().size() == 0) {
+            return new CommandResult(String.format(MESSAGE_NO_EMPLOYEE, tmrDate.getDayOfWeek()));
+        }
+        return new CommandResult(String.format(MESSAGE_SUCCESS, model.getFilteredPersonList().size(),
+                tmrDate.getDayOfWeek()));
+    }
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TomorrowCommand.java]: Same comments as TodayCommand

File: src/main/java/seedu/address/logic/commands/TomorrowCommand.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+
+import seedu.address.model.Model;
+
+
```
Comment (ureshiiYing) [inline src/main/java/seedu/address/logic/commands/TomorrowCommand.java]: Missing javadoc description

File: src/main/java/seedu/address/logic/commands/TodayCommand.java (hunk 4 above)
Comment (ureshiiYing) [reply src/main/java/seedu/address/logic/commands/TodayCommand.java]: It's also missing a javadoc description

**Review by ureshiiYing — COMMENTED** [review]
Other comments: Perhaps naming the commands as `TodayCommand` and `TomorrowCommand` isn't the best? It felt vague to me. The missing javadoc descriptions did not help too.

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java (hunk 1 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TagFindCommand.java]: Indeed, I forgot to change the javadoc after editing from the original "c-find" code. Thanks! I've edited this part.

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java (hunk 2 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TagFindCommand.java]: Thank you for the suggestion. Yes, I think that sounds clearer. I've changed it to `FindByTagCommand`. Thanks!

File: src/main/java/seedu/address/logic/commands/TagFindCommand.java (hunk 3 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TagFindCommand.java]: I've changed it in the latest commit, thanks!

File: src/main/java/seedu/address/logic/commands/TodayCommand.java (hunk 4 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TodayCommand.java]: `c-tag-find` was implemented on top of  `c-today` feature, so `c-tag-find` branch contains the `c-today` feature.

Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TodayCommand.java]: I've added the javadoc. Thanks!

File: src/main/java/seedu/address/logic/commands/TodayCommand.java (hunk 5 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TodayCommand.java]: Removed, thanks!

File: src/main/java/seedu/address/logic/commands/TodayCommand.java (hunk 6 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TodayCommand.java]: Added a new line. Thanks!

File: src/main/java/seedu/address/logic/commands/TomorrowCommand.java (hunk 8 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TomorrowCommand.java]: Added in the latest commit. thanks!

File: src/main/java/seedu/address/logic/commands/TomorrowCommand.java (hunk 7 above)
Comment (yanbingtao) [reply src/main/java/seedu/address/logic/commands/TomorrowCommand.java]: Added new lines in between in the latest commit. Thank you!

**Comment by yanbingtao** [comment]
> Other comments: Perhaps naming the commands as `TodayCommand` and `TomorrowCommand` isn't the best? It felt vague to me. The missing javadoc descriptions did not help too.

I've changed these two to `FindByTagTodayCommand.java` and `FindByTagTomorrowCommand.java`. And two Javadoc were also added for these two classes to better explain the meaning. Thanks!

**Review by Diwu-Yi — DISMISSED** [review]
LGTM

**Review by Diwu-Yi — APPROVED** [review]
LGTM

**Review by Persdre — APPROVED** [review]
LGTM

**Pull request merged**